gourmet/gtk_extras/LinkedTextView.py

```diff
@@ -97,22 +69,28 @@ def __init__ (self):
         self.connect('motion-notify-event',self.motion_notify_event)
         self.connect('visibility-notify-event',self.visibility_notify_event)
 
-    def make_buffer (self):
+    def make_buffer(self):
         return LinkedPangoBuffer()
 
-    # Links can be activated by pressing Enter.
-    def key_press_event(self, text_view, event):
+    def key_press_event(self,
+                        text_view: 'LinkedTextView',
+                        event: Gdk.Event) -> bool:
+        """Handle Enter key-press on time links."""
         keyname = Gdk.keyval_name(event.keyval)
-        if keyname in ['Return','KP_Enter']:
+        if keyname in ['Return', 'KP_Enter']:
             buffer = text_view.get_buffer()
-            iter = buffer.get_iter_at_mark(buffer.get_insert())
-            return self.follow_if_link(text_view, iter)
+            itr = buffer.get_iter_at_mark(buffer.get_insert())
+            return self.follow_if_link(text_view, itr)
         return False
-    # Links can also be activated by clicking.
-    def event_after(self, text_view, event):
+
+    def event_after(self,
+                    text_view: 'LinkedTextView',
+                    event: Gdk.Event) -> bool:
+        """Handle mouse clicks on time links."""
         if event.type != Gdk.EventType.BUTTON_RELEASE:
             return False
-        if event.button != 1:
+        _, button = event.get_button()
+        if button != 1:  # is not the left-button (as set by the system, not hw)
```

Comment:
I would prefer that being formulated in the positive:

`if event.type == Gdk.EventType.BUTTON_RELEASE and event.get_button()[1] == 1:`

That would make it easier if we ever were to add other interactions. But that's just taste.